src/js/review/SubmitController.jsx

```diff
@@ -51,35 +64,36 @@ class SubmitController extends React.Component {
       trackingPrefix
     } = this.props;
 
-    let isValid;
-    let errors;
+    const preSubmit = this.getPreSubmit(formConfig);
+    let isValid = true;
 
-    // If a pre-submit agreement was specified, it has to be accepted first
-    const preSubmitField = formConfig.preSubmitInfo &&
-        formConfig.preSubmitInfo.required && (formConfig.preSubmitInfo.field || 'AGREED');
-    if (preSubmitField && !form.data[preSubmitField]) {
+    // If a pre-submit agreement is required, make sure it was accepted
+    if (preSubmit.required && !form.data[preSubmit.field]) {
```

Comment:
This if/else block is a bit confusing to me. The logic I see is the following:
```
if preSubmit is required AND is not true in the form data
  isValid = false
else (it's either not required OR (it is required AND it is true in the form data))
  run the isValidForm check and set isValid to whatever the output is
  
  if isValid is false
    record and track information about it

if isValid is true
  submit the form
else (isValid is false)
  set an attempted submit status
```
So it seems like there are 2 main if/else blocks here:
1. The first is used to set the value of `isValid`
2. The second is used to do things based on the value of `isValid`

What I think is a bit confusing is there's logic within the first block that does things based on the value of `isValid` (tracking events). Would it be clearer if that was contained in the second block?